src/porepy/geometry/domain.py

```diff
@@ -0,0 +1,374 @@
+"""Module containing classes and functions for definining and manupilating a domain."""
+
+from __future__ import annotations
+
+from typing import NamedTuple, Optional
+
+import numpy as np
+import numpy.typing as npt
+
+import porepy as pp
+
+
+class Domain:
+    """Class for the geometrical representation of the domain.
+
+    There are two ways of constructing a domain in PorePy:
+
+    - (1) By passing the bounding box as a dictionary (we assume therefore that the
+      domain is a box) or,
+
+    - (2) By passing a polytope (polygon in 2d and poyhedron in 3d), which are lists of
+      numpy arrays defining a general domain (this also includes non-convex domains).
+
+    See the class constructor documentation for more details.
```

Comment:
Please move __init__ documentation to the class documentation. See How-to-docstring for details.

Reply:
Done.